tests/parsing/FilterHelperTest.py

```diff
@@ -0,0 +1,52 @@
+import unittest
+
+from coalib.parsing.FilterHelper import (
+    apply_filter,
+    apply_filters,
+    is_valid_filter,
+    _filter_section_bears
+)
+from coalib.parsing.filters import available_filters
+from coalib.parsing.InvalidFilterException import InvalidFilterException
+from coalib.settings.ConfigurationGathering import get_all_bears
+
+
+class FilterHelperTest(unittest.TestCase):
+
+    def test_apply_filter_exception(self):
+        with self.assertRaises(InvalidFilterException) as exp:
+            apply_filter('unknown', ['args'])
+
+        message = str(exp.exception)
+        self.assertEqual("'unknown' is an invalid filter. Available filters: "
+                         + ', '.join(sorted(available_filters)), message)
+
+    def test_apply_filters_exception(self):
+        with self.assertRaises(InvalidFilterException) as exp:
+            apply_filters([['language', 'C', 'Python'],
+                           ['unknown', 'args']])
+
+        message = str(exp.exception)
+        self.assertEqual("'unknown' is an invalid filter. Available filters: "
+                         + ', '.join(sorted(available_filters)), message)
+
+    def test_is_valid_filter(self):
+        result1 = is_valid_filter('check')
+        result2 = is_valid_filter('can_detect')
+        self.assertFalse(result1)
+        self.assertTrue(result2)
```

Comment:
instead of using `result1` maybe we could use`resultForCheck`. Or instead why not use `self.assertFalse(is_valid_filter('check'))`

Reply:
Saving the output in a variable increases readability IMO.